kahuna/public/js/services/image/usages.js

```diff
@@ -0,0 +1,60 @@
+import angular from 'angular';
+import Immutable from 'immutable';
+
+import Rx from 'rx';
+
+export const imageUsagesService = angular.module('gr.image-usages.service', ['kahuna.edits.service']);
+
+
+imageUsagesService.factory('imageUsagesService', [function() {
+
+    function forImage(imageResource) {
+        function usageTitle(usage) {
+            const referenceType =
+                usage.get('platform') == 'print' ? 'indesign' : 'frontend';
+
+            const build = (usage, referenceType) => {
+                const reference = usage.get('references').find(u =>
+                    u.get('type') == referenceType);
+
+                return (reference && reference.get('name')) ? reference.get('name') : 'No title found.';
+            };
+
+            return build(usage, referenceType);
```

Comment:
Is the extra `build` function needed/helping if you're calling it immediately anyway?